Spec/Auth.swift

```diff
@@ -845,6 +845,37 @@ class Auth : QuickSpec {
             }
         }
 
+        // RSA8f3
+        it("ensure the message published with a wildcard '*' does not have a clientId") {
+            let options = AblyTests.commonAppSetup()
+            // Request a token with a wildcard '*' value clientId
+            options.token = getTestToken(clientId: "*")
+            let rest = ARTRest(options: options)
+            rest.httpExecutor = mockExecutor
+            let channel = rest.channels.get("test")
+
+            waitUntil(timeout: testTimeout) { done in
+                let message = ARTMessage(name: nil, data: "no client")
+                expect(message.clientId).to(beNil())
+                channel.publish([message]) { error in
+                    expect(error).to(beNil())
+                    switch extractBodyAsMessages(mockExecutor.requests.first) {
+                    case .Failure(let error):
+                        fail(error)
+                    case .Success(let httpBody):
+                        expect(httpBody.unbox.first!["clientId"]).to(beNil())
+                    }
+                    channel.history { page, error in
+                        expect(error).to(beNil())
+                        expect(page!.items).to(haveCount(1))
+                        expect((page!.items[0] as! ARTMessage).clientId).to(beNil())
+                        done()
+                    }
+                }
+            }
+            expect(rest.auth.clientId).to(beNil())
```

Comment:
@tcard Then this is wrong https://github.com/ably/ably-ios/blob/master/Source/ARTAuth.m#L346? I saw [RSA12](http://docs.ably.io/client-lib-development-guide/features/#RSA12) and it should be nil.  The nil _value indicates that the client has not yet been able to confirm its identity_. I think RSA8f3 spec is wrong. WDYT?